### ISSUE TITLE
Unexpected key(s) in state_dict: "transformer.text_encoder.embeddings.position_ids". 

### ISSUE THREAD

**Issue by vdorbala** [ashkamath/mdetr#106]
When running the demo script, I get - 
`Unexpected key(s) in state_dict: "transformer.text_encoder.embeddings.position_ids".`

while trying to load the model using -
`model, postprocessor = torch.hub.load('ashkamath/mdetr:main', 'mdetr_efficientnetB5', pretrained=True, return_postprocessor=True)`

Does the model not work anymore?

**Comment by iremeyiokur** [comment]
It might be related with the transformers version. I solved the problem by adding `strict=False` into `load_state_dict` as proposed in the following link which basically ignores non-matching keys. I hope it will work for you.

https://discuss.pytorch.org/t/missing-keys-unexpected-keys-in-state-dict-when-loading-self-trained-model/22379/6

**Comment by Buchimi** [comment]
How can this be done in the colab?

**Comment by SKALGO22** [comment]
> It might be related with the transformers version. I solved the problem by adding `strict=False` into `load_state_dict` as proposed in the following link which basically ignores non-matching keys. I hope it will work for you.
> 
> https://discuss.pytorch.org/t/missing-keys-unexpected-keys-in-state-dict-when-loading-self-trained-model/22379/6

@iremeyiokur , your proposed fix, loads the model without errors, but then I get no logits while making inferences...it's just a NaN tensor. I also tried deleting that particular key from the `sate_dict` yet the issue persists. Was it the same for you?
